src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -3,13 +3,35 @@
 import core.basesyntax.Storage;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private final Pair<K, V>[] storage;
+
+    public StorageImpl() {
+        storage = new Pair[10];
+    }
 
     @Override
     public void put(K key, V value) {
+        for (int i = 0; i < storage.length; i++) {
+            if (storage[i] == null) {
+                storage[i] = Pair.of(key, value);
+                break;
+            }
+            if (storage[i].getFirst() == key) {
```

Comment:
```suggestion
            if (storage[i].getFirst() == key) {
```
What if it's not the same object, but equal to it?